src/new-contract-interaction/new-contract-interaction.js

```diff
@@ -0,0 +1,163 @@
+    alertId: 'AE-AAVE-NEW-CONTRACT-INTERACTION',
+    severity: FindingSeverity.Medium,
+    type: FindingType.Suspicious,
+    metadata: {
+      address,
+      contractAge,
+    },
+    everestId: AAVE_EVEREST_ID,
+  });
+}
+
+function provideHandleTransaction(ethersProvider) {
+  return async function handleTransaction(txEvent) {
+    const findings = [];
+
+    // for performance reasons, don't continue to run this handler if an Etherscan API key
+    // was not provided
+    if (apiKey === undefined) {
+      return findings;
+    }
+
+    // get all addresses involved with this transaction that are non-Aave contracts
+    let addresses = Object.keys(txEvent.addresses);
+
+    // to minimize Etherscan requests (slow and rate limited to 5/sec for the free tier),
+    // exclude the lending pool address, the incentives controller, and all token addresses
+    let exclusions = [
+      lendingPoolV2Address,
+      contractAddresses.IncentivesController.toLowerCase(),
+    ];
+    exclusions = exclusions.concat(tokenAddresses);
+    addresses = addresses.filter((item) => exclusions.indexOf(item) < 0);
+
+    // watch for recently created contracts interacting with Aave lending pool
+    if (txEvent.transaction.to === lendingPoolV2Address) {
+      // create an array of promises that retrive the contract code for each address
+      const contractCodePromises = [];
+      let contractCode = [];
+      addresses.forEach((address) => {
+        // RPC call for contract code
+        const codePromise = ethersProvider.getCode(address);
+
+        // associate each address with the code that was returned
+        codePromise.then((result) => contractCode.push({ address, code: result }));
+
+        // to prevent Promise.all() from rejecting, catch failed promises and set the return
+        // value to undefined
+        contractCodePromises.push(codePromise.catch(() => undefined));
+      });
+
+      // resolve the requests
+      await Promise.all(contractCodePromises);
+
+      // filter out EOAs from our original list of addresses
+      contractCode = contractCode.filter((item) => (item.code !== '0x'));
+      addresses = contractCode.map((item) => item.address);
+
+      // Next, for each contract, query the Etherscan API for the list of transactions
+      // the first transaction item returned in the results will be the earliest
+      const etherscanTxlistPromises = [];
+      const txData = [];
+      contractCode.forEach((item) => {
+        const txlistPromise = axios.get(baseUrl + item.address + options + apiKey);
+
+        // associate each address with the transaction list that was returned
+        txlistPromise.then((result) => txData.push({ address: item.address, response: result }));
+
+        // to prevent Promise.all() from rejecting, catch failed promises and set the return
+        // value to undefined
+        etherscanTxlistPromises.push(txlistPromise.catch(() => undefined));
+      });
+
+      // resolve the requests
```

Comment:
Rabbit hole: I think what we're waiting for here is for the promises to be `settled`.